src/plugins/vis_type_table/public/components/table_vis_component_group.tsx

```diff
@@ -22,16 +16,16 @@ export const TableVisComponentGroup = memo(
   ({ tableGroups, visConfig, event, uiState }: TableVisGroupComponentProps) => {
     return (
       <>
-        {tableGroups.map(({ table, title }) => (
-          <EuiFlexItem key={title} className="visTable__group">
+        {tableGroups.map(({ table, title }, index) => (
+          <div key={title} className="visTable__group" data-test-subj={`visTableGroup${index}`}>
```

Comment:
This element should remain `EuiFlexItem`. OUI (or EUI, as it is referred to in Dashboards) is our component library, and we want to make sure we're using it throughout the application. `EuiFlexItem` supports the `data-test-subj` prop, it should just be a matter of adding it.